Source/JavaScriptCore/runtime/MathObject.cpp

```diff
@@ -246,27 +246,35 @@ JSC_DEFINE_HOST_FUNCTION(mathProtoFuncHypot, (JSGlobalObject* globalObject, Call
     }, NulloptBehavior::Abort);
     RETURN_IF_EXCEPTION(scope, { });
 
-    double max = 0;
+    bool hasNaN = false;
+    double maxAbs = 0;
+    double sum = 0;
     for (double argument : args) {
-        if (std::isinf(argument))
+        if (std::isinf(argument)) [[unlikely]]
             return JSValue::encode(jsDoubleNumber(+std::numeric_limits<double>::infinity()));
-        max = std::max(std::abs(argument), max);
+        if (std::isnan(argument)) [[unlikely]] {
+            hasNaN = true;
+            continue;
+        }
+
+        double absArgument = std::fabs(argument);
+        if (maxAbs < absArgument) {
+            double scaledArgument = maxAbs / absArgument;
+            sum = std::fma(sum, scaledArgument * scaledArgument, 1);
+            maxAbs = absArgument;
+        } else if (0 < maxAbs) {
```

Comment:
When can this case happen?

Reply:
This happens with cases like `Math.hypot(4, 3, 2, 1)` where the current value is smaller than `maxAbs`.